### PR TITLE
First round of changes proposed by mission team

### PR DESCRIPTION
**Pull request by cgobat**
## Summary
This PR mostly contains the changes that were discussed at SBN's last migration meeting on Feb 1. Commit messages should hopefully provide a fairly clear picture of what was updated, but to summarize:
- Adds a new `nh:Engineering_Conversion_Coeffcients` class and associated attributes for handling cases where polynomial coefficients are currently provided in the text description of table columns in the PDS3 labels. For example: [swa_0340156832_0x586_sci.lbl](https://pds-smallbodies.astro.umd.edu/holdings/nh-x-swap-3-kemcruise1-v1.0/data/20161030_034015/swa_0340156832_0x586_sci.lbl) (Ctrl+F search for "coefficients")
- Updates the `nh:MVIC_Conversion_Constants` class name to `nh:Radiometric_Conversion_Constants`, since other instruments besides MVIC use the same set of conversion constants
- Adds a more complete set of permissible detector names and types within the `nh:Detector/nh:detector_name` and `nh:Detector/nh:detector_type` attributes
  - Also adds a `nh:SWAP_Details` (to mirror the existing `nh:MVIC_Details`) class as a child of `nh:Detector` to hold a `nh:sweep_samples_count` attribute
- Adds an optional `nh:arrokoth_constant` attribute to `nh:Resolved_Source` and `nh:Unresolved_Source`
- Adds a couple more permissible `nh:mission_phase_name`s: "KEM1 Cruise" and "KEM2 Cruise"

## Test Data and/or Report
All tests passing after updates.

## Related Issues
Blocks #12.

### CODE REVIEW

File: docs/source/detailed/outline.rst
Comment (acraugh) [inline docs/source/detailed/outline.rst:1]: The change makes sense for the new class, but the old class will have to be re-inserted with a notation that it is deprecated. This is a document change, so I'll worry about it. It'll give me a chance to play around with formatting and see what works best for showing a deprecated feature.

File: docs/source/examples/mc0procMockup.rst
Comment (acraugh) [inline docs/source/examples/mc0procMockup.rst:1]: Another file where the deprecated MVIC_Calibration_Constants class needs to be indicated. I'll deal with this in the documentation update branch (to be created).

File: docs/source/examples/mpfprocMockup.rst
Comment (acraugh) [inline docs/source/examples/mpfprocMockup.rst:1]: MVIC_Conversion_Constants again. Same solution.

File: docs/source/examples/swaprocMockup.rst
Comment (acraugh) [inline docs/source/examples/swaprocMockup.rst:1]: System file paths like "../../../build/development/*/1.19.0.0/PDS4_NH_1J00_1100.xsd" need to be expunged in examples. Too many users copy them, and this is very bad practice for submitted label files. I'll do a more detailed review as part of the document merge and clean up details like this.

File: docs/source/user/user-guide.rst
Comment (acraugh) [inline docs/source/user/user-guide.rst:1]: More documentation. I'll have to think about what to do with this as well...

File: src/PDS4_NH_IngestLDD.xml
Comment (acraugh) [inline src/PDS4_NH_IngestLDD.xml:1]: Couple things, here:
* I'm wondering how well-known "SSD" is as an acronym for "Solid State Silicon Detector", which I would expect to be called an "SSSD".
*  This is a classic example of a circular definition: "The engineering_component_description attribute gives a description of the named engineering component."  There are a few of these in here. Can we add a little more detail? I haven't come across a definition yet that tells me what an "engineering component" might be as a concept, for example. Assume someone who is reading these definitions is sufficiently confused or naive about the data they are looking at that they don't know anything about the mission, instrument, target, or data collection. They need a little extra context to make sense of definitions that might be perfectly fine in the context of a table in a SIS.

File: test/MVICmc0Proc_VALID.xml
Comment (acraugh) [inline test/MVICmc0Proc_VALID.xml:1]: Interesting. The deprecation needs to be tested, so the original version of the test needs to remain but it now becomes a...warn case? Hm. I'll ask Jordan about that one. Then, of course, we'll need a version with the new structure.

As a general rule, existing regressions tests should NEVER be edited for a change - their job is to be static to prove there have been no expected changes.

File: test/MVICmpfProc_VALID.xml
Comment (acraugh) [inline test/MVICmpfProc_VALID.xml:1]: Previous file comment applies here as well.

File: test/SWAPhistProc_VALID.xml
Comment (acraugh) [inline test/SWAPhistProc_VALID.xml:1]: This one's fine, I assume (I didn't poke into details), but where's the usual complementary FAIL test? Are there no Schematron rules to check?

Ah - I see. I never made the FAIL tests for MVIC. Shame on me! OK, this one can be my problem, too...

**Review by acraugh — COMMENTED** [review]
Left a bunch of notes, but I think I need/want to take care of most of them because they're part of the documentation site, and I still need to develop some workflows for that.

File: src/PDS4_NH_IngestLDD.xml
Comment (cgobat) [reply src/PDS4_NH_IngestLDD.xml:1]: @katecrombie any thoughts on SSD? I have no personal preference and would be happy to change that to whatever.

File: src/PDS4_NH_IngestLDD.xml [hunk 1]
```diff
@@ -114,12 +197,47 @@
...
       </DD_Permissible_Value>
+      <DD_Permissible_Value>
+        <value>Pluto Energetic Particles Spectrometer Science Investigation</value>
+        <value_meaning>
+          Data were obtained using the PEPSSI time-of-flight (TOF) spectrometer that
+          measures ions and electrons over a broad range of energies and pitch angles.
+        </value_meaning>
+      </DD_Permissible_Value>
+      <DD_Permissible_Value>
+        <value>Radio Science Expriment</value>
```
Comment (cgobat) [inline src/PDS4_NH_IngestLDD.xml]: Typo here to be fixed

File: src/PDS4_NH_IngestLDD.xml
Comment (katecrombie) [reply src/PDS4_NH_IngestLDD.xml:1]: I had to look up the SSD definition, and had the same thought as Anne did, but the instrument team used "SSD" .  Not really helpful, I know.  Best I can do for now is say that as we get additional feedback from the teams we will update definitions.

**Comment by acraugh** [comment]
Every other value is spelled out except for "CCD", which seems to be
well-known , but I would actually prefer to see spelled out as well for
consistency and clarity that will better stand up over time. These labels
are here to support future users, not the mission team, after all.
Including both is also a fine option, as in "Charge-Coupled Device (CCD)".
Explicit is good.

Are these being copied verbatim from some source, or is it relatively easy
to make the change to the spelled-out value? I view it as highly desirable,
but at some point practicality enters the equation.

-Anne.

File: src/PDS4_NH_IngestLDD.xml (hunk 1 above)
Comment (acraugh) [reply src/PDS4_NH_IngestLDD.xml]: Done.

**Comment by acraugh** [comment]
I'd like to merge this PR (#11)  this morning, if possible. The doc and regression tests can wait for cleanup. The big question is this: Can I change the Permissible_Values "CCD" to "Charge-Coupled Device (CCD)" and "SSD" to "Solid State Silicon Detector (SSD)"? I would note that "SSD" commonly stands for "Solid State Drive" (as opposed to "HDD") and has for some years, so if this is something other than an SSD, and it sounds like it is, then it definitely needs words rather than letters alone.  

Will this cause a major disruption on your end, @cgobat?

**Comment by cgobat** [comment]
Nope, both those changes work for me. Thanks!

**Pull request merged**